File: app/lib/services/api_service.dart
import 'dart:convert';
import 'dart:io';
import 'package:http/http.dart' as http;
import 'package:flutter_secure_storage/flutter_secure_storage.dart';
import 'package:jwt_decoder/jwt_decoder.dart';
import '../config/app_config.dart';

class ApiService {
  static final ApiService _instance = ApiService._internal();
  factory ApiService() => _instance;
  ApiService._internal();

  final FlutterSecureStorage _secureStorage = const FlutterSecureStorage();
  static const String _tokenKey = 'auth_token';
  static const String _refreshTokenKey = 'refresh_token';

  // Get auth headers
  Future<Map<String, String>> _getHeaders({bool includeAuth = true}) async {
    final headers = <String, String>{
      'Content-Type': 'application/json',
      'Accept': 'application/json',
    };

    if (includeAuth) {
      final token = await _secureStorage.read(key: _tokenKey);
      if (token != null) {
        // Check if token is expired
        if (JwtDecoder.isExpired(token)) {
          await _refreshToken();
          final newToken = await _secureStorage.read(key: _tokenKey);
          if (newToken != null) {
            headers['Authorization'] = 'Bearer $newToken';
          }
        } else {
          headers['Authorization'] = 'Bearer $token';
        }
      }
    }

    return headers;
  }

  // Refresh token
  Future<void> _refreshToken() async {
    try {
      final refreshToken = await _secureStorage.read(key: _refreshTokenKey);
      if (refreshToken == null) throw Exception('No refresh token');

      final response = await http.post(
        Uri.parse('${AppConfig.apiBaseUrl}${AppConfig.auth}/refresh'),
        headers: {'Content-Type': 'application/json'},
        body: jsonEncode({'refreshToken': refreshToken}),
      );

      if (response.statusCode == 200) {
        final data = jsonDecode(response.body);
        await _secureStorage.write(key: _tokenKey, value: data['token']);
        if (data['refreshToken'] != null) {
          await _secureStorage.write(key: _refreshTokenKey, value: data['refreshToken']);
        }
      } else {
        throw Exception('Token refresh failed');
      }
    } catch (e) {
      await logout();
      rethrow;
    }
  }

  // Generic HTTP request method
  Future<http.Response> _makeRequest(
    String method,
    String endpoint, {
    Map<String, dynamic>? body,
    Map<String, String>? queryParams,
    bool includeAuth = true,
  }) async {
    final uri = Uri.parse('${AppConfig.apiBaseUrl}$endpoint');
    final finalUri = queryParams != null ? uri.replace(queryParameters: queryParams) : uri;
    final headers = await _getHeaders(includeAuth: includeAuth);

    late http.Response response;

    try {
      switch (method.toUpperCase()) {
        case 'GET':
          response = await http.get(finalUri, headers: headers).timeout(
            AppConfig.connectionTimeout,
          );
          break;
        case 'POST':
          response = await http.post(
            finalUri,
            headers: headers,
            body: body != null ? jsonEncode(body) : null,
          ).timeout(AppConfig.connectionTimeout);
          break;
        case 'PUT':
          response = await http.put(
            finalUri,
            headers: headers,
            body: body != null ? jsonEncode(body) : null,
          ).timeout(AppConfig.connectionTimeout);
          break;
        case 'DELETE':
          response = await http.delete(finalUri, headers: headers).timeout(
            AppConfig.connectionTimeout,
          );
          break;
        default:
          throw UnsupportedError('HTTP method $method is not supported');
      }
    } on SocketException {
      throw Exception('No internet connection');
    } on HttpException {
      throw Exception('Server error');
    } catch (e) {
      throw Exception('Network error: $e');
    }

    return _handleResponse(response);
  }

  // Handle response
  http.Response _handleResponse(http.Response response) {
    switch (response.statusCode) {
      case 200:
      case 201:
        return response;
      case 400:
        throw Exception('Bad request: ${response.body}');
      case 401:
        logout(); // Clear tokens on unauthorized
        throw Exception('Unauthorized: Please login again');
      case 403:
        throw Exception('Forbidden: Access denied');
      case 404:
        throw Exception('Not found');
      case 409:
        throw Exception('Conflict: ${response.body}');
      case 500:
        throw Exception('Internal server error');
      default:
        throw Exception('Error: ${response.statusCode} - ${response.body}');
    }
  }

  // Authentication methods
  Future<Map<String, dynamic>> login(String email, String password) async {
    final response = await _makeRequest(
      'POST',
      '${AppConfig.auth}/login',
      body: {'email': email, 'password': password},
      includeAuth: false,
    );

    final data = jsonDecode(response.body);
    await _secureStorage.write(key: _tokenKey, value: data['token']);
    if (data['refreshToken'] != null) {
      await _secureStorage.write(key: _refreshTokenKey, value: data['refreshToken']);
    }
    return data;
  }

  Future<Map<String, dynamic>> register(Map<String, dynamic> userData) async {
    final response = await _makeRequest(
      'POST',
      '${AppConfig.auth}/register',
      body: userData,
      includeAuth: false,
    );

    final data = jsonDecode(response.body);
    await _secureStorage.write(key: _tokenKey, value: data['token']);
    if (data['refreshToken'] != null) {
      await _secureStorage.write(key: _refreshTokenKey, value: data['refreshToken']);
    }
    return data;
  }

  Future<void> logout() async {
    try {
      await _makeRequest('POST', '${AppConfig.auth}/logout');
    } catch (e) {
      // Continue with local logout even if server logout fails
    } finally {
      await _secureStorage.delete(key: _tokenKey);
      await _secureStorage.delete(key: _refreshTokenKey);
    }
  }

  Future<bool> isLoggedIn() async {
    final token = await _secureStorage.read(key: _tokenKey);
    return token != null && !JwtDecoder.isExpired(token);
  }

  // Salon methods
  Future<List<Map<String, dynamic>>> getSalons({Map<String, String>? filters}) async {
    final response = await _makeRequest('GET', AppConfig.salons, queryParams: filters);
    final data = jsonDecode(response.body);
    return List<Map<String, dynamic>>.from(data['salons'] ?? data);
  }

  Future<Map<String, dynamic>> getSalon(String salonId) async {
    final response = await _makeRequest('GET', '${AppConfig.salons}/$salonId');
    return jsonDecode(response.body);
  }

  Future<List<Map<String, dynamic>>> getSalonServices(String salonId) async {
    final response = await _makeRequest('GET', '${AppConfig.salons}/$salonId${AppConfig.services}');
    final data = jsonDecode(response.body);
    return List<Map<String, dynamic>>.from(data['services'] ?? data);
  }

  Future<List<Map<String, dynamic>>> getSalonAvailability(String salonId, String date) async {
    final response = await _makeRequest(
      'GET',
      '${AppConfig.salons}/$salonId/availability',
      queryParams: {'date': date},
    );
    final data = jsonDecode(response.body);
    return List<Map<String, dynamic>>.from(data['availability'] ?? data);
  }

  // Booking methods
  Future<Map<String, dynamic>> createBooking(Map<String, dynamic> bookingData) async {
    final response = await _makeRequest(
      'POST',
      AppConfig.bookings,
      body: bookingData,
    );
    return jsonDecode(response.body);
  }

  Future<List<Map<String, dynamic>>> getUserBookings() async {
    final response = await _makeRequest('GET', AppConfig.bookings);
    final data = jsonDecode(response.body);
    return List<Map<String, dynamic>>.from(data['bookings'] ?? data);
  }

  Future<Map<String, dynamic>> updateBookingStatus(String bookingId, String status) async {
    final response = await _makeRequest(
      'PUT',
      '${AppConfig.bookings}/$bookingId/status',
      body: {'status': status},
    );
    return jsonDecode(response.body);
  }

  Future<void> cancelBooking(String bookingId) async {
    await _makeRequest('DELETE', '${AppConfig.bookings}/$bookingId');
  }

  // Payment methods
  Future<Map<String, dynamic>> createPaymentIntent(Map<String, dynamic> paymentData) async {
    final response = await _makeRequest(
      'POST',
      '${AppConfig.payments}/intent',
      body: paymentData,
    );
    return jsonDecode(response.body);
  }

  Future<Map<String, dynamic>> confirmPayment(String paymentIntentId) async {
    final response = await _makeRequest(
      'POST',
      '${AppConfig.payments}/confirm',
      body: {'paymentIntentId': paymentIntentId},
    );
    return jsonDecode(response.body);
  }

  // User methods
  Future<Map<String, dynamic>> getUserProfile() async {
    final response = await _makeRequest('GET', AppConfig.users);
    return jsonDecode(response.body);
  }

  Future<Map<String, dynamic>> updateUserProfile(Map<String, dynamic> userData) async {
    final response = await _makeRequest('PUT', AppConfig.users, body: userData);
    return jsonDecode(response.body);
  }

  // Favorites methods
  Future<void> addToFavorites(String salonId) async {
    await _makeRequest('POST', '${AppConfig.users}/favorites', body: {'salonId': salonId});
  }

  Future<void> removeFromFavorites(String salonId) async {
    await _makeRequest('DELETE', '${AppConfig.users}/favorites/$salonId');
  }

  Future<List<Map<String, dynamic>>> getFavorites() async {
    final response = await _makeRequest('GET', '${AppConfig.users}/favorites');
    final data = jsonDecode(response.body);
    return List<Map<String, dynamic>>.from(data['favorites'] ?? data);
  }

  // Notifications methods
  Future<List<Map<String, dynamic>>> getNotifications() async {
    final response = await _makeRequest('GET', AppConfig.notifications);
    final data = jsonDecode(response.body);
    return List<Map<String, dynamic>>.from(data['notifications'] ?? data);
  }

  Future<void> markNotificationAsRead(String notificationId) async {
    await _makeRequest('PUT', '${AppConfig.notifications}/$notificationId/read');
  }

  Future<void> markAllNotificationsAsRead() async {
    await _makeRequest('PUT', '${AppConfig.notifications}/read-all');
  }

  // FCM Token methods
  Future<void> updateFCMToken(String token) async {
    await _makeRequest('POST', '${AppConfig.users}/fcm-token', body: {'token': token});
  }
}

File: salon-booking-portal/lib/api-service.ts
import axios, { AxiosInstance, AxiosResponse } from 'axios'
import Cookies from 'js-cookie'

// Environment configuration
const API_BASE_URL = process.env.NEXT_PUBLIC_API_URL || 'https://api.blushrz.com/api'
const ENVIRONMENT = process.env.NEXT_PUBLIC_ENV || 'production'

// API endpoints
export const API_ENDPOINTS = {
  AUTH: {
    LOGIN: '/auth/admin/login',
    LOGOUT: '/auth/admin/logout',
    REFRESH: '/auth/admin/refresh',
    ME: '/auth/admin/me',
  },
  SALONS: {
    LIST: '/admin/salons',
    CREATE: '/admin/salons',
    UPDATE: '/admin/salons',
    DELETE: '/admin/salons',
    GET: (id: string) => `/admin/salons/${id}`,
    SERVICES: (id: string) => `/admin/salons/${id}/services`,
    STAFF: (id: string) => `/admin/salons/${id}/staff`,
    AVAILABILITY: (id: string) => `/admin/salons/${id}/availability`,
    UPDATE_STATUS: (id: string) => `/admin/salons/${id}/status`,
  },
  SERVICES: {
    LIST: '/admin/services',
    CREATE: '/admin/services',
    UPDATE: '/admin/services',
    DELETE: '/admin/services',
    GET: (id: string) => `/admin/services/${id}`,
  },
  STAFF: {
    LIST: '/admin/staff',
    CREATE: '/admin/staff',
    UPDATE: '/admin/staff',
    DELETE: '/admin/staff',
    GET: (id: string) => `/admin/staff/${id}`,
  },
  BOOKINGS: {
    LIST: '/admin/bookings',
    CREATE: '/admin/bookings',
    UPDATE: '/admin/bookings',
    DELETE: '/admin/bookings',
    GET: (id: string) => `/admin/bookings/${id}`,
    UPDATE_STATUS: (id: string) => `/admin/bookings/${id}/status`,
    BY_DATE: '/admin/bookings/by-date',
    BY_SALON: (id: string) => `/admin/bookings/salon/${id}`,
  },
  USERS: {
    LIST: '/admin/users',
    GET: (id: string) => `/admin/users/${id}`,
    UPDATE: (id: string) => `/admin/users/${id}`,
    FAVORITES: (id: string) => `/admin/users/${id}/favorites`,
  },
  PAYMENTS: {
    LIST: '/admin/payments',
    GET: (id: string) => `/admin/payments/${id}`,
    REFUND: (id: string) => `/admin/payments/${id}/refund`,
    WEBHOOK_LOGS: '/admin/payments/webhook-logs',
  },
  ANALYTICS: {
    OVERVIEW: '/admin/analytics/overview',
    BOOKINGS: '/admin/analytics/bookings',
    REVENUE: '/admin/analytics/revenue',
    SALONS: '/admin/analytics/salons',
    SERVICES: '/admin/analytics/services',
    USERS: '/admin/analytics/users',
    EXPORT: '/admin/analytics/export',
  },
  NOTIFICATIONS: {
    LIST: '/admin/notifications',
    CREATE: '/admin/notifications',
    SEND: '/admin/notifications/send',
    UPDATE: (id: string) => `/admin/notifications/${id}`,
    DELETE: (id: string) => `/admin/notifications/${id}`,
  },
  OFFERS: {
    LIST: '/admin/offers',
    CREATE: '/admin/offers',
    UPDATE: '/admin/offers',
    DELETE: '/admin/offers',
    GET: (id: string) => `/admin/offers/${id}`,
  },
}

// Token management
const TOKEN_KEY = 'admin_token'
const REFRESH_TOKEN_KEY = 'admin_refresh_token'

class ApiService {
  private api: AxiosInstance
  private refreshPromise: Promise<string> | null = null

  constructor() {
    this.api = axios.create({
      baseURL: API_BASE_URL,
      timeout: 30000,
      headers: {
        'Content-Type': 'application/json',
        'Accept': 'application/json',
      },
    })

    this.setupInterceptors()
  }

  private setupInterceptors() {
    // Request interceptor - Add auth token
    this.api.interceptors.request.use(
      (config) => {
        const token = this.getToken()
        if (token) {
          config.headers.Authorization = `Bearer ${token}`
        }
        return config
      },
      (error) => Promise.reject(error)
    )

    // Response interceptor - Handle token refresh
    this.api.interceptors.response.use(
      (response) => response,
      async (error) => {
        const originalRequest = error.config

        if (error.response?.status === 401 && !originalRequest._retry) {
          originalRequest._retry = true

          try {
            const newToken = await this.refreshToken()
            originalRequest.headers.Authorization = `Bearer ${newToken}`
            return this.api(originalRequest)
          } catch (refreshError) {
            this.logout()
            return Promise.reject(refreshError)
          }
        }

        return Promise.reject(error)
      }
    )
  }

  // Token management
  private getToken(): string | null {
    if (typeof window === 'undefined') return null
    return Cookies.get(TOKEN_KEY) || localStorage.getItem(TOKEN_KEY)
  }

  private setToken(token: string): void {
    if (typeof window === 'undefined') return
    Cookies.set(TOKEN_KEY, token, { expires: 7, secure: true, sameSite: 'strict' })
    localStorage.setItem(TOKEN_KEY, token)
  }

  private getRefreshToken(): string | null {
    if (typeof window === 'undefined') return null
    return Cookies.get(REFRESH_TOKEN_KEY) || localStorage.getItem(REFRESH_TOKEN_KEY)
  }

  private setRefreshToken(token: string): void {
    if (typeof window === 'undefined') return
    Cookies.set(REFRESH_TOKEN_KEY, token, { expires: 30, secure: true, sameSite: 'strict' })
    localStorage.setItem(REFRESH_TOKEN_KEY, token)
  }

  private clearTokens(): void {
    if (typeof window === 'undefined') return
    Cookies.remove(TOKEN_KEY)
    Cookies.remove(REFRESH_TOKEN_KEY)
    localStorage.removeItem(TOKEN_KEY)
    localStorage.removeItem(REFRESH_TOKEN_KEY)
  }

  private async refreshToken(): Promise<string> {
    if (this.refreshPromise) {
      return this.refreshPromise
    }

    this.refreshPromise = this.performTokenRefresh()
    
    try {
      const token = await this.refreshPromise
      return token
    } finally {
      this.refreshPromise = null
    }
  }

  private async performTokenRefresh(): Promise<string> {
    const refreshToken = this.getRefreshToken()
    if (!refreshToken) {
      throw new Error('No refresh token available')
    }

    try {
      const response = await axios.post(
        `${API_BASE_URL}${API_ENDPOINTS.AUTH.REFRESH}`,
        { refreshToken },
        { timeout: 10000 }
      )

      const { token, refreshToken: newRefreshToken } = response.data
      this.setToken(token)
      if (newRefreshToken) {
        this.setRefreshToken(newRefreshToken)
      }

      return token
    } catch (error) {
      this.clearTokens()
      throw error
    }
  }

  // Authentication methods
  async login(email: string, password: string) {
    try {
      const response: AxiosResponse = await this.api.post(API_ENDPOINTS.AUTH.LOGIN, {
        email,
        password,
      })

      const { token, refreshToken, user } = response.data
      this.setToken(token)
      if (refreshToken) {
        this.setRefreshToken(refreshToken)
      }

      return { user, token }
    } catch (error) {
      throw this.handleError(error)
    }
  }

  async logout() {
    try {
      await this.api.post(API_ENDPOINTS.AUTH.LOGOUT)
    } catch (error) {
      // Continue with logout even if server request fails
    } finally {
      this.clearTokens()
      // Redirect to login page
      if (typeof window !== 'undefined') {
        window.location.href = '/login'
      }
    }
  }

  async getCurrentUser() {
    try {
      const response: AxiosResponse = await this.api.get(API_ENDPOINTS.AUTH.ME)
      return response.data
    } catch (error) {
      throw this.handleError(error)
    }
  }

  // Generic HTTP methods
  private async request<T>(method: 'GET' | 'POST' | 'PUT' | 'DELETE', url: string, data?: any): Promise<T> {
    try {
      const config = {
        method,
        url,
        ...(data && { data }),
      }

      const response: AxiosResponse<T> = await this.api.request(config)
      return response.data
    } catch (error) {
      throw this.handleError(error)
    }
  }

  // Salons
  async getSalons(params?: any) {
    return this.request('GET', API_ENDPOINTS.SALONS.LIST, params)
  }

  async getSalon(id: string) {
    return this.request('GET', API_ENDPOINTS.SALONS.GET(id))
  }

  async createSalon(data: any) {
    return this.request('POST', API_ENDPOINTS.SALONS.CREATE, data)
  }

  async updateSalon(id: string, data: any) {
    return this.request('PUT', API_ENDPOINTS.SALONS.GET(id), data)
  }

  async deleteSalon(id: string) {
    return this.request('DELETE', API_ENDPOINTS.SALONS.GET(id))
  }

  async updateSalonStatus(id: string, status: string) {
    return this.request('PUT', API_ENDPOINTS.SALONS.UPDATE_STATUS(id), { status })
  }

  // Services
  async getServices(params?: any) {
    return this.request('GET', API_ENDPOINTS.SERVICES.LIST, params)
  }

  async getService(id: string) {
    return this.request('GET', API_ENDPOINTS.SERVICES.GET(id))
  }

  async createService(data: any) {
    return this.request('POST', API_ENDPOINTS.SERVICES.CREATE, data)
  }

  async updateService(id: string, data: any) {
    return this.request('PUT', API_ENDPOINTS.SERVICES.GET(id), data)
  }

  async deleteService(id: string) {
    return this.request('DELETE', API_ENDPOINTS.SERVICES.GET(id))
  }

  // Staff
  async getStaff(params?: any) {
    return this.request('GET', API_ENDPOINTS.STAFF.LIST, params)
  }

  async getStaffMember(id: string) {
    return this.request('GET', API_ENDPOINTS.STAFF.GET(id))
  }

  async createStaffMember(data: any) {
    return this.request('POST', API_ENDPOINTS.STAFF.CREATE, data)
  }

  async updateStaffMember(id: string, data: any) {
    return this.request('PUT', API_ENDPOINTS.STAFF.GET(id), data)
  }

  async deleteStaffMember(id: string) {
    return this.request('DELETE', API_ENDPOINTS.STAFF.GET(id))
  }

  // Bookings
  async getBookings(params?: any) {
    return this.request('GET', API_ENDPOINTS.BOOKINGS.LIST, params)
  }

  async getBooking(id: string) {
    return this.request('GET', API_ENDPOINTS.BOOKINGS.GET(id))
  }

  async createBooking(data: any) {
    return this.request('POST', API_ENDPOINTS.BOOKINGS.CREATE, data)
  }

  async updateBooking(id: string, data: any) {
    return this.request('PUT', API_ENDPOINTS.BOOKINGS.GET(id), data)
  }

  async deleteBooking(id: string) {
    return this.request('DELETE', API_ENDPOINTS.BOOKINGS.GET(id))
  }

  async updateBookingStatus(id: string, status: string) {
    return this.request('PUT', API_ENDPOINTS.BOOKINGS.UPDATE_STATUS(id), { status })
  }

  // Analytics
  async getAnalyticsOverview(params?: any) {
    return this.request('GET', API_ENDPOINTS.ANALYTICS.OVERVIEW, params)
  }

  async getBookingsAnalytics(params?: any) {
    return this.request('GET', API_ENDPOINTS.ANALYTICS.BOOKINGS, params)
  }

  async getRevenueAnalytics(params?: any) {
    return this.request('GET', API_ENDPOINTS.ANALYTICS.REVENUE, params)
  }

  async getSalonsAnalytics(params?: any) {
    return this.request('GET', API_ENDPOINTS.ANALYTICS.SALONS, params)
  }

  async getServicesAnalytics(params?: any) {
    return this.request('GET', API_ENDPOINTS.ANALYTICS.SERVICES, params)
  }

  async getUsersAnalytics(params?: any) {
    return this.request('GET', API_ENDPOINTS.ANALYTICS.USERS, params)
  }

  // Users
  async getUsers(params?: any) {
    return this.request('GET', API_ENDPOINTS.USERS.LIST, params)
  }

  async getUser(id: string) {
    return this.request('GET', API_ENDPOINTS.USERS.GET(id))
  }

  async updateUser(id: string, data: any) {
    return this.request('PUT', API_ENDPOINTS.USERS.UPDATE(id), data)
  }

  // Payments
  async getPayments(params?: any) {
    return this.request('GET', API_ENDPOINTS.PAYMENTS.LIST, params)
  }

  async getPayment(id: string) {
    return this.request('GET', API_ENDPOINTS.PAYMENTS.GET(id))
  }

  async refundPayment(id: string, reason?: string) {
    return this.request('POST', API_ENDPOINTS.PAYMENTS.REFUND(id), { reason })
  }

  // Notifications
  async getNotifications(params?: any) {
    return this.request('GET', API_ENDPOINTS.NOTIFICATIONS.LIST, params)
  }

  async createNotification(data: any) {
    return this.request('POST', API_ENDPOINTS.NOTIFICATIONS.CREATE, data)
  }

  async sendNotification(data: any) {
    return this.request('POST', API_ENDPOINTS.NOTIFICATIONS.SEND, data)
  }

  // Offers
  async getOffers(params?: any) {
    return this.request('GET', API_ENDPOINTS.OFFERS.LIST, params)
  }

  async getOffer(id: string) {
    return this.request('GET', API_ENDPOINTS.OFFERS.GET(id))
  }

  async createOffer(data: any) {
    return this.request('POST', API_ENDPOINTS.OFFERS.CREATE, data)
  }

  async updateOffer(id: string, data: any) {
    return this.request('PUT', API_ENDPOINTS.OFFERS.GET(id), data)
  }

  async deleteOffer(id: string) {
    return this.request('DELETE', API_ENDPOINTS.OFFERS.GET(id))
  }

  // Error handling
  private handleError(error: any): Error {
    if (error.response) {
      // Server responded with error status
      const message = error.response.data?.message || 'Server error occurred'
      const status = error.response.status

      switch (status) {
        case 401:
          return new Error('Authentication failed. Please log in again.')
        case 403:
          return new Error('Access denied. Insufficient permissions.')
        case 404:
          return new Error('Resource not found.')
        case 409:
          return new Error(message || 'Conflict occurred.')
        case 422:
          return new Error(message || 'Invalid data provided.')
        case 500:
          return new Error('Internal server error. Please try again later.')
        default:
          return new Error(message || `Request failed with status ${status}`)
      }
    } else if (error.request) {
      // Network error
      return new Error('Network error. Please check your connection.')
    } else {
      // Other error
      return new Error(error.message || 'An unexpected error occurred.')
    }
  }

  // Utility methods
  isAuthenticated(): boolean {
    return !!this.getToken()
  }

  getApiUrl(): string {
    return API_BASE_URL
  }

  getEnvironment(): string {
    return ENVIRONMENT
  }
}

// Create singleton instance
export const apiService = new ApiService()
export default apiService
